### PR TITLE
Napari fix

### PR DESCRIPTION
**Pull request by LucaMarconato**
This pr fixed bugs that appeared when testing the recently merged io branch on the datasets in `spatialdata-sandbox` and when visualizing them with `napari_spatialdata (branch spatialdata)`.

In particular:
- fixed how transformations are created by the parsers
- points are stored as `pyarrow.Table` instead of `GeoDataFrame` due to performance limitations
- fixed minor things, like improved docs

### CODE REVIEW

**Comment by LucaMarconato** [comment]
mmm how does this pr seem so big, this branch was built on top of `feature/transform_and_model`, I'll find out what's happening.

**Comment by ivirshup** [comment]
Looked at the git history a bit, and 😨

If it were me I would strongly consider:

* Create a new branch
* Diff this branch against main
* Copy over the intentional changes from this branch to the new branch

Otherwise, if you could apply the same squash that happened to the other PR, that could make this easier. I'm not sure how possible that would be if there were merge conflicts that got resolved while making this PR.

**Comment by LucaMarconato** [comment]
Thank Isaac, yes gonna do as you described. Merge this in a different branch and then remerge to main so that it will be just a few files changed.

**Comment by LucaMarconato** [comment]
Fixed the mess and now it's ready for review!

I have done the merge carefully (took 1 hour) and all the tests here and `spatialdata-sandbox` pass. I hope not to have missed anything.

File: spatialdata/_io/read.py [hunk 1]
```diff
@@ -210,28 +218,30 @@ def _read_shapes(store: Union[str, Path, MutableMapping, zarr.Group], fmt: Spati
...
-    geo_df = GeoDataFrame({"geometry": geometry}, index=index)
-    for c in f["annotations"]:
-        column = read_elem(f["annotations"][c])
-        geo_df[c] = column
+    # geometry = from_ragged_array(typ, coords)
+    #
+    # geo_df = GeoDataFrame({"geometry": geometry}, index=index)
+    # for c in f["annotations"]:
+    #     column = read_elem(f["annotations"][c])
+    #     geo_df[c] = column
```
Comment (kevinyamauchi) [inline spatialdata/_io/read.py:241]: I think we should just remove the commented out code.

File: spatialdata/_io/read.py [hunk 2]
```diff
@@ -210,28 +218,30 @@ def _read_shapes(store: Union[str, Path, MutableMapping, zarr.Group], fmt: Spati
...
-    index = np.array(f["Index"])
+    path = os.path.join(f._store.path, f.path, "points.parquet")
+    table = pq.read_table(path)
+    # coords = np.array(f["coords"])
+    # index = np.array(f["Index"])
     # offsets_keys = [k for k in f.keys() if k.startswith("offset")]
     # offsets = tuple(np.array(f[k]).flatten() for k in offsets_keys)
 
-    typ = fmt.attrs_from_dict(f.attrs.asdict())
+    # typ = fmt.attrs_from_dict(f.attrs.asdict())
```
Comment (kevinyamauchi) [inline spatialdata/_io/read.py:232]: I think we should just remove the commented out code.

File: tests/_core/test_models.py [hunk 3]
```diff
@@ -86,9 +87,9 @@ def test_points_model(
         coords = RNG.normal(size=(10, 2))
         if annotations is not None:
             annotations["A"] = annotations["A"].astype(str)
-        points = model.parse(coords, annotations)
-        assert PointsModel.GEOMETRY_KEY in points
-        assert PointsModel.TRANSFORM_KEY in points.attrs
+        points = model.parse(coords, None if annotations is None else pa.Table.from_pandas(annotations))
+        # assert PointsModel.GEOMETRY_KEY in points
```
Comment (kevinyamauchi) [inline tests/_core/test_models.py:91]:
```suggestion
```

File: spatialdata/_io/read.py (hunk 1 above)
Comment (kevinyamauchi) [reply spatialdata/_io/read.py:241]:
```suggestion
```

File: spatialdata/_io/read.py [hunk 4]
```diff
@@ -210,28 +218,30 @@ def _read_shapes(store: Union[str, Path, MutableMapping, zarr.Group], fmt: Spati
...
+) -> pa.Table:
     """Read points from a zarr store."""
     f = zarr.open(store, mode="r")
 
-    coords = np.array(f["coords"])
-    index = np.array(f["Index"])
+    path = os.path.join(f._store.path, f.path, "points.parquet")
+    table = pq.read_table(path)
+    # coords = np.array(f["coords"])
+    # index = np.array(f["Index"])
```
Comment (kevinyamauchi) [inline spatialdata/_io/read.py:228]:
```suggestion
```

File: spatialdata/_io/read.py (hunk 2 above)
Comment (kevinyamauchi) [reply spatialdata/_io/read.py:232]:
```suggestion
```

**Review by kevinyamauchi — APPROVED** [review]
This looks fine to me. I made a few suggestions for deleting unused code below. Thanks!

**Comment by LucaMarconato** [comment]
Thanks @kevinyamauchi for the review. In the next prs (one already opened 😜) I will be more descriptive. In this case, this pr is fixing all the bugs discovered when trying to visualize the data with napari, or when trying to convert the data in spatialdata-sandbox. The big pr merged to main was not tested on those two external repos.

(I accepted your changes one by one, actually it would have been better to do a batch commit, gonna do it the next time!)

**Pull request merged**